packages/aws-cdk-lib/aws-iam/lib/oidc-provider.ts

```diff
@@ -138,6 +140,8 @@ export class OpenIdConnectProvider extends Resource implements IOpenIdConnectPro
   public constructor(scope: Construct, id: string, props: OpenIdConnectProviderProps) {
     super(scope, id);
 
+    const rejectUnauthorized = FeatureFlags.of(this).isEnabled(IAM_OIDC_REJECT_UNAUTHORIZED_CONNECTIONS) ?? false;
```

Comment:
can we add unit tests to https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-iam/test/oidc-provider.test.ts for checking cases when this flag is enabled/disabled?